addon/pages/devtoolsPanel.html

```diff
@@ -3,19 +3,129 @@
   <head>
     <meta charset="UTF-8">
     <link rel="stylesheet" type="text/css" href="chrome://browser/content/extension.css">
+    <style>
+      .hidden {
+        display: none !important;
+      }
+
+      body {
+        padding: 10px;
+      }
+
+      h1 {
+        margin: 0 0 10px 0;
+        font-size: 100%;
+      }
+
+      .box {
+        border: 1px solid silver;
+        padding: 10px;
+        margin-bottom: 10px;
+      }
+
+      #no-selection, #no-labels {
+        font-style: italic;
+      }
+
+      #has-selection label {
+        width: 50em;
+      }
+
+      #labels table {
+        border-spacing: 0;
+        padding: 0;
+      }
+
+      #labels .path, #labels .label, #labels .action {
+        white-space: nowrap;
+      }
+
+      #labels .preview {
+        width: 100%;
+      }
+
+      #labels td.preview {
+        font-family: monospace;
+        font-size: 90%;
+        max-height: 1em;
+      }
+
+      #labels tr.hover:hover {
+        background-color: #eee;
+      }
+
+      #labels .action-button {
+        color: #000;
+        text-decoration: none;
+        width: 1.5em;
+        display: inline-block;
+        text-align: center;
+      }
+
+      #labels .action-button:hover {
+        color: rebeccapurple;
+        outline: 1px solid #444;
+      }
+
+      #labels th, #labels td {
+        padding: 2px 5px;
+      }
+
+      #spinner-container {
+        display: inline-block;
+        color: #888;
+      }
+
+      #spinner {
+        display: inline-block;
+        width: 1em;
+        height: 1em;
+        margin: 0 8px;
+      }
+
+      #spinner:after {
+        content: " ";
+        display: block;
+        width: 0.5em;
+        height: 0.5em;
+        margin: 1px;
+        border-radius: 50%;
+        border: 5px solid #888;
+        border-color: #888 transparent #888 transparent;
+        animation: spinner 1.2s linear infinite;
+      }
+
+      @keyframes spinner {
+        0% {
+          transform: rotate(0deg);
+        }
+        100% {
+          transform: rotate(360deg);
+        }
+      }
+
+    </style>
   </head>
-  <body class="browser-style" style="padding: 0 20px">
-    <p>
-      Use the devtools Inspector to select an element, then label it here. Once you have labeled all the interesting elements, save.
-    <p>
-    <div>
-      <button id="labelButton">Label the inspected element</button>
-      as
-      <input id="labelField" type="text">
+
+  <body class="browser-style">
+    <div id="current-labels" class="box">
+      <div id="no-selection" class="hidden">
+        No currently selected element - use the DevTools <b>Inspector</b> to select an element.
```

Comment:
Might as well replace the hyphen with the more typographically correct em dash: —. Spaces around it are a matter for style guides, but I generally go without in proportional fonts.